src/librepgp/stream-parse.c

```diff
@@ -603,7 +616,8 @@ encrypted_src_read_aead_part(pgp_source_encrypted_param_t *param)
         }
     }
 
-    res = encrypted_start_aead_chunk(param, chunkend ? param->chunkidx + 1 : param->chunkidx, lastchunk);
+    res = encrypted_start_aead_chunk(
+      param, chunkend && param->chunkin ? param->chunkidx + 1 : param->chunkidx, lastchunk);
```

Comment:
I think some parentheses around the ternary conditional expr. would make this a bit clearer.

Reply:
Thanks. Replaced it with additional variable and if, seems to look a bit better.